.buildkite/pipeline.yml

```diff
@@ -0,0 +1,31 @@
+steps:
+  - label: ":docker: Publish to ECR"
+    command: |
+      echo "~~~ :key: Logging In"
+      aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/automattic
+      echo "~~~ :hammer: Building Image"
+      docker build -t android-build-image .
+      # If a tag is present use it
+      if [ -n "$BUILDKITE_TAG" ]; then
+        echo "~~~ :label: Tagging Image with version $BUILDKITE_TAG"
+        docker tag android-build-image:latest public.ecr.aws/automattic/android-build-image:$BUILDKITE_TAG
+        echo "~~~ :arrow_up: Pushing Image"
+        docker push public.ecr.aws/automattic/android-build-image:$BUILDKITE_TAG
+        exit 0
+      fi
+      if [ $BUILDKITE_BRANCH = "trunk" ]; then
```

Comment:
@jkmassel The issue with `$ABC` variables getting substituted before they are run means that this comparison ends up being:

```
if [ add/buildkite = "trunk" ]; then
```

I am starting to wonder if we should always put our Buildkite actions in bash scripts and call them so we don't have to deal with this weird(?) substitution issue.

Reply:
Yeah, that is kinda weird, but IMHO isn't a huge deal because it still evaluates correctly when it's:

```bash
if [ trunk = "trunk" ]; then
```

I'd be fine to put it in a script (it'd make our `.yml` files prettier, but I'm not sure what it's gaining us tbh?

Reply:
I didn't mean the script bit for this PR - I think we should move this publishing logic to a plugin, so I am treating this pipeline as if it's temporary.

I think we should consider using a script for all our Buildkite builds to side-step the substitution issue as it's very easy to miss and end up costing us debug time. Happened to me several times.